fix: update UPGRADING.md (#8274)
vegaprotocol/vega: UPGRADING.md

> Upgrading from v0.53.0 to v0.71.4

Reviewer: I believe they're actually upgrading from 0.53.1

Author: Its actually 0.53.2. Changed now.

=================================

> This document covers the list of major changes between versions 0.53.0 and 0.71.4. For specific settings needed to be done for each component,
> please refer to <DOCS_URL>/**mainnet**/node-operators/migration-guides/upgrade-mainnet-to-0714.

Reviewer: ```suggestion
This document covers the list of major changes between versions 0.53.0 and 0.71.4. For instructions on how to complete the upgrade,
please refer to https://docs.vega.xyz/mainnet/node-operators/migration-guides.
```
NOTE that URL doesn't work yet

Author: Okay.

Upgrading from v0.67.1 to v0.71.4
=================================

Read through for a list of the major changes between versions 0.67.1 and 0.71.4, covering:
* [Configuration changes](#configuration-changes)
* [Command line changes](#command-line-changes)

# Configuration changes

> This document should cover most of the breaking change in using the command line or configuring the different software. We also recommend that node operator tries to generate a bare configuration (`vega init...`, `vega datanode init...`) in a temporary folder and compare this new configuration to their existing ones to address any changes. Any missing entries in the configuration will be using defaults.

Reviewer: ```suggestion
This document covers most of the breaking changes in using the command line or configuring the different softwares. 

We also recommend that node operator generates a bare configuration (`vega init...`, `vega datanode init...`) in a temporary folder to compare this new configuration to their existing ones to address any changes. Any missing entries in the configuration will be using defaults.
```

## Vega

The Vega configuration file can be found in `$VEGA_HOME/config/node/config.toml`.

## Data node

The data node configuration file can be found in `$DATANODE_HOME/config/data-node/config.toml`.

### Settings added in 0.71.4

> **_Rate Limiting_** - In the past the datanode API did rate limiting only by the number of subscriptions open per IPs. With the new version a rate limit per IP was introduced. The server now allows 20 API calls per second on average with a burst of up to 100 api calls in a short period of time.
> For details how to use this setting and configure this functionality, please refer to [rateLimiting](https://github.com/vegaprotocol/vega/blob/develop/datanode/ratelimit/README.md)
> 
> **_REST and GraphQL configuration_** - Previously the configuration of the Graphql and REST APIs used separate HTTP servers and configuration. This was merge onto a single server now. Also the default endpoint for Graphql changed from /query to /graphql
> Usage example:
> ```Toml
> [Gateway]
>   Port = 3008
>   IP = "0.0.0.0"
>   [Gateway.GraphQL]
>     Enabled = true
>     Endpoint = "/graphql"
>   [Gateway.REST]
>     Enabled = true
> ```
> 
> **_IPFS update_** - The ipfs library used by the datanode to share network history have been updated. This may require manual steps to reset the state and apply the ipfs migrations.

Reviewer: ```suggestion
**_Rate Limiting_** - In the past, the data node API rate limited based only on the number of subscriptions open per IP. In the new version, a rate limit per IP was introduced. The server now allows 20 API calls per second on average with a burst of up to 100 API calls in a short period of time.

For details how to use this setting and configure this functionality, please refer to [rateLimiting](https://github.com/vegaprotocol/vega/blob/develop/datanode/ratelimit/README.md)

**_REST and GraphQL configuration_** - Previously the configuration of the GraphQL and REST APIs used separate HTTP servers and configuration. This was merged into a single server. 

Also, the default endpoint for GraphQL has changed from /query to /graphql.

Usage example:
```Toml
[Gateway]
  Port = 3008
  IP = "0.0.0.0"
  [Gateway.GraphQL]
    Enabled = true
    Endpoint = "/graphql"
  [Gateway.REST]
    Enabled = true
```

**_IPFS update_** - The IPFS library used by the data node to share network history has been updated. This may require manual steps to reset the state and apply the IPFS migrations.
```

Upgrading from v0.67.1 to v0.71.4
=================================

Read through for a list of the major changes between versions 0.67.1 and 0.71.4, covering:
* [Configuration changes](#configuration-changes)
* [Command line changes](#command-line-changes)

# Configuration changes

This document covers most of the breaking changes in using the command line or configuring the different softwares. 

> We also recommend that node operator generates a bare configuration (`vega init...`, `vega datanode init...`) in a temporary folder to compare this new configuration to their existing ones to address any changes. Any missing entries in the configuration will be using defaults.

Reviewer: ```suggestion
We also recommend that node operators generate a bare configuration (`vega init...`, `vega datanode init...`) in a temporary folder to compare this new configuration to their existing ones to address any changes. Any missing entries in the configuration will be using defaults.
```

Reviewer: sorry, missed this first time around

## Vega

The Vega configuration file can be found in `$VEGA_HOME/config/node/config.toml`.

## Data node

The data node configuration file can be found in `$DATANODE_HOME/config/data-node/config.toml`.

### Settings added in 0.71.4

> **_Rate Limiting_** - In the past the datanode API did rate limiting only by the number of subscriptions open per IPs. With the new version a rate limit per IP on the unary calls for the RESY, GraphQL and grpc APIs was introduced. The server now allows 20 API calls per second on average with a burst of up to 100 api calls in a short period of time.

Reviewer: ```suggestion
**_Rate Limiting_** - In the past, the data node API did rate limiting only by the number of subscriptions open per IP. In the new version, a rate limit per IP on the unary calls for the REST, GraphQL and gRPC APIs was introduced. The server now allows 20 API calls per second on average with a burst of up to 100 API calls in a short period of time.
```

> For details how to use this setting and configure this functionality, please refer to [rateLimiting](https://github.com/vegaprotocol/vega/blob/develop/datanode/ratelimit/README.md)

Reviewer: ```suggestion
For details how to use this setting and configure this functionality, please refer to the [rate limiting](https://github.com/vegaprotocol/vega/blob/develop/datanode/ratelimit/README.md) documentation.
```

> **_REST and GraphQL configuration_** - Previously the configuration of the Graphql and REST APIs used separate HTTP servers and configuration. This was merge onto a single server now. Also the default endpoint for Graphql changed from /query to /graphql

Reviewer: ```suggestion
**_REST and GraphQL configuration_** - Previously the configuration of the GraphQL and REST APIs used separate HTTP servers and configuration. This has now been merged into a single server. 

Also, the default endpoint for GraphQL has changed from /query to /graphql.
```

Usage example:
```Toml
[Gateway]
  Port = 3008
  IP = "0.0.0.0"
  [Gateway.GraphQL]
    Enabled = true
    Endpoint = "/graphql"
  [Gateway.REST]
    Enabled = true
```

> **_IPFS update_** - The ipfs library used by the datanode to share network history have been updated. This may require manual steps to reset the state and apply the ipfs migrations.

Reviewer: ```suggestion
**_IPFS update_** - The IPFS library used by the data node to share network history has been updated. This may require manual steps to reset the state and apply the IPFS migrations.
```

### Settings removed in 0.71.4

> **_X-Vega-Connection_** - This HTTP header is now deprecated from API calls to the datanode.

Reviewer: ```suggestion
**_X-Vega-Connection_** - This HTTP header is now deprecated from API calls to the data node.
```

## Visor

### Breaking changes with 0.71.4

> Visor used to require both the vega and datanode binaries. With recent changes, the vega binary now embed the datanode cli as a subcommands. This has been reflected to the visor configuration as now a single binary is required in the configuration.

Reviewer: ```suggestion
Visor used to require both the Vega and data node binaries. With recent changes, the Vega binary is now embedded in the data node CLI as a subcommand. This has been reflected in the Visor configuration as now only a single binary is required in the configuration.
```

> Visor used to require both the vega and datanode binaries. With recent changes, the vega binary is now embedding the datanode cli as a subcommands. This has been reflected to the visor configuration as now a single binary is required in the configuration.

Reviewer: ```suggestion
Visor used to require both the Vega and data node binaries. With recent changes, the Vega binary now has the data node CLI embedded in it. This has been reflected in the Visor configuration, as now only a single binary is required in the configuration.
```

> Here's an example of visor configuration reflecting these changes:

Reviewer: ```suggestion
Here's an example of Visor configuration reflecting these changes:
```

```Toml
maxNumberOfFirstConnectionRetries = 43200
maxNumberOfRestarts = 3
restartsDelaySeconds = 5
stopDelaySeconds = 3
stopSignalTimeoutSeconds = 15

[upgradeFolders]
  "vX.X.X" = "vX.X.X"

[autoInstall]
  enabled = true
  repositoryOwner = "vegaprotocol"
  repository = "vega"
  [autoInstall.assets]
    assset_name = "vega-linux-amd64.zip"
    binary_name = "vega"
```

# Command line changes

## Vega

#### Deprecations

> `IssueSignatures` is no longer a validator command and is now protected by the spam engine.

Reviewer: `spam engine` may be better phrased as "spam protection" or "spam protection engine"? for the sake of consistency, as I don't think it's called the spam engine anywhere I've seen it described.

### Settings removed in 0.71.4

**_X-Vega-Connection_** - This HTTP header is now deprecated from API calls to the data node.

## Visor

### Breaking changes with 0.71.4

Visor used to require both the Vega and data node binaries. With recent changes, the Vega binary is now embedded in the data node CLI as a subcommand. This has been reflected in the Visor configuration as now only a single binary is required in the configuration.

Here's an example of Visor configuration reflecting these changes:
```Toml
maxNumberOfFirstConnectionRetries = 43200
maxNumberOfRestarts = 3
restartsDelaySeconds = 5
stopDelaySeconds = 3
stopSignalTimeoutSeconds = 15

[upgradeFolders]
  "vX.X.X" = "vX.X.X"

[autoInstall]
  enabled = true
  repositoryOwner = "vegaprotocol"
  repository = "vega"
  [autoInstall.assets]
    assset_name = "vega-linux-amd64.zip"
    binary_name = "vega"
```

# Command line changes

## Vega

#### Deprecations

`IssueSignatures` is no longer a validator command and is now protected by the spam protection engine.

> ### VegaWallet

Reviewer: ```suggestion
### Vega Wallet
```

### Settings added in 0.71.4

> **_Rate Limiting_** - In the past the datanode API did rate limiting only by the number of subscriptions open per IPs. With the new version a rate limit per IP on the unary calls for the RESY, GraphQL and grpc APIs was introduced. The server now allows 20 API calls per second on average with a burst of up to 100 api calls in a short period of time.

Author: on the unary calls for the RESY, - should be REST

Reviewer 2: thanks. addressed.

```

## New service

> With version 0.71.4, `vegavisor` was introduced to help facilitate protocol upgrades. This tool is not required to run the node but recommended in order to ease software upgrades when they're expected by the network. Read more about [Vega Visor](https://github.com/vegaprotocol/vega/blob/develop/visor/readme.md).

Reviewer: Perhaps this section needs rewording to introduce it without tying it to a version because validators have used it with testnet versions older than 0.71 so the edit is technically not right.

Author: Yes, corrected that.
Thanks.
